shared/pkg/analyzer/trivy/trivy.go

```diff
@@ -0,0 +1,125 @@
+// Copyright © 2022 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trivy
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/aquasecurity/trivy/pkg/commands/artifact"
+	trivyFlag "github.com/aquasecurity/trivy/pkg/flag"
+
+	"github.com/openclarity/kubeclarity/shared/pkg/analyzer"
+	"github.com/openclarity/kubeclarity/shared/pkg/config"
+	"github.com/openclarity/kubeclarity/shared/pkg/formatter"
+	"github.com/openclarity/kubeclarity/shared/pkg/job_manager"
+	"github.com/openclarity/kubeclarity/shared/pkg/utils"
+)
+
+const AnalyzerName = "trivy"
+
+type Analyzer struct {
+	name       string
+	logger     *log.Entry
+	config     config.AnalyzerTrivyConfigEx
+	resultChan chan job_manager.Result
+	localImage bool
+}
+
+func New(conf *config.Config,
+	logger *log.Entry,
+	resultChan chan job_manager.Result,
+) job_manager.Job {
+	return &Analyzer{
+		name:       AnalyzerName,
+		logger:     logger.Dup().WithField("analyzer", AnalyzerName),
+		config:     config.CreateAnalyzerTrivyConfigEx(conf.Analyzer, conf.Registry),
+		resultChan: resultChan,
+		localImage: conf.LocalImageScan,
+	}
+}
+
+func (a *Analyzer) Run(sourceType utils.SourceType, userInput string) error {
+	a.logger.Infof("Called %s analyzer on source %v %v", a.name, sourceType, userInput)
+	go func() {
+		res := &analyzer.Results{}
+
+		var output bytes.Buffer
+		trivyOptions := trivyFlag.Options{
+			GlobalOptions: trivyFlag.GlobalOptions{
+				Timeout: a.config.Timeout,
+			},
+			ScanOptions: trivyFlag.ScanOptions{
+				Target:         userInput,
+				SecurityChecks: nil, // Disable all security checks for SBOM only scan
+			},
+			ReportOptions: trivyFlag.ReportOptions{
+				Format:       "cyclonedx", // Cyconedx format for SBOM so that we don't need to convert
+				ReportFormat: "all",       // Full report not just summary
+				Output:       &output,     // Save the output to our local buffer instead of Stdout
+				ListAllPkgs:  true,        // List all packages for full SBOM
```

Comment:
Needed to look what this means for trivy. maybe better to state "List all packages regardless of vulnerabilities"

Reply:
updated the comment